cmd/dot_env.go

```diff
@@ -0,0 +1,102 @@
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path"
+	"runtime"
+	"strings"
+
+	"github.com/mitchellh/cli"
+	"trellis-cli/trellis"
+)
+
+type DotEnvCommand struct {
+	UI      cli.Ui
+	Trellis *trellis.Trellis
+}
+
+func (c *DotEnvCommand) Run(args []string) int {
+	if err := c.Trellis.LoadProject(); err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+
+	var environment string
+
+	switch len(args) {
+	case 0:
+		environment = "development"
+	case 1:
+		environment = args[0]
+	default:
+		c.UI.Error(fmt.Sprintf("Error: too many arguments (expected 0 or 1, got %d)\n", len(args)))
+		c.UI.Output(c.Help())
+		return 1
+	}
+
+	_, ok := c.Trellis.Environments[environment]
+	if !ok {
+		c.UI.Error(fmt.Sprintf("Error: %s is not a valid environment", environment))
+		return 1
+	}
+
+	// Copy playbook file from package to Trellis
+	_, filename, _, ok := runtime.Caller(0)
+	playbookTemplatePath := path.Join(path.Dir(filename), "../playbooks/dotenv.yml")
+
+	b, readFileErr := ioutil.ReadFile(playbookTemplatePath)
+	if readFileErr != nil {
+		log.Fatal(readFileErr)
+	}
+
+	playbookPath := "dotenv.yml"
+	writeFileErr := ioutil.WriteFile(playbookPath, b, 0644)
+	if writeFileErr != nil {
+		log.Fatal(writeFileErr)
+	}
+
+	dotEnv := execCommand("ansible-playbook", "dotenv.yml", "-e", "env=" + environment)
+
+	env := os.Environ()
+	// To allow mockExecCommand injects its environment variables
+	if dotEnv.Env != nil {
+		env = dotEnv.Env
+	}
+	dotEnv.Env = append(env, "ANSIBLE_RETRY_FILES_ENABLED=false")
+
+	logCmd(dotEnv, c.UI, true)
+	runErr := dotEnv.Run()
+
+	// Remove playbook file no matter command success or not
+	removeFileErr := os.Remove(playbookPath)
```

Comment:
I think we'd want this to be done via `defer` for more guarantees:

```go
defer deletePlaybook(playbookPath) # function can do error handling
```

Reply:
done.